### PR TITLE
feat: cdk8s Argo Workflows TDE-904

### PR DESCRIPTION
**Pull request by paulfouquet**
#### Motivation

Deploy Argo Workflows using its Helm chart via CDK8s

#### Modification

Namespace and service accounts are managed in `aws-cdk`.
Helm Chart is deployed with CDK8s.

#### Checklist

_If not applicable, provide explanation of why._

- [ ] Tests updated - No tests for this feature
- [x] Docs updated
- [x] Issue linked in Title

### CODE REVIEW

File: infra/cdk8s.ts [hunk 1]
```diff
@@ -45,6 +49,12 @@ async function main(): Promise<void> {
 
   karpenterProvisioner.addDependency(karpenter);
 
+  new ArgoWorkflows(app, 'argo-workflows', {
+    clusterName: ClusterName,
```
Comment (blacha) [inline infra/cdk8s.ts:52]:
```suggestion
    clusterName,
```

File: infra/charts/argo.workflows.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  /** Name of the EKS cluster
+   *
+   * @example "Workflows"
+   */
+  clusterName: string;
+}
+
+/** This is the version of the Helm chart for Argo Workflows
+ *
+ * (Do not mix up with Argo Workflows application version)
```
Comment (blacha) [inline infra/charts/argo.workflows.ts:26]: Would be great to link to the helm chart git repo or where this number can be found!

File: infra/charts/argo.workflows.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      'eks.amazonaws.com/capacityType': 'ON_DEMAND',
+      'kubernetes.io/arch': 'amd64',
+    };
+
+    new Helm(this, 'argo-workflows', {
+      chart: 'argo-workflows',
+      releaseName: 'argo-workflows',
+      repo: 'https://argoproj.github.io/argo-helm',
+      namespace: 'argo',
+      version: version,
```
Comment (blacha) [inline infra/charts/argo.workflows.ts:60]:
```suggestion
      version,
```

File: infra/eks/cluster.ts [hunk 4]
```diff
@@ -203,20 +203,24 @@ export class LinzEksCluster extends Stack {
       new PolicyStatement({ actions: ['logs:PutRetentionPolicy'], resources: ['*'], effect: Effect.ALLOW }),
     );
     fluentBitSa.node.addDependency(fluentBitNs); // Ensure the namespace created first
-
     new CfnOutput(this, CfnOutputKeys.FluentBit.ServiceAccountName, { value: fluentBitSa.serviceAccountName });
 
-    // Basic constructs for argo to be deployed into
+    // Argo Workflows
```
Comment (blacha) [inline infra/eks/cluster.ts:208]: Could we link this to the cdk8s argo?

Comment (paulfouquet) [reply infra/eks/cluster.ts:208]: It does not seem possible with vscode built in features.

**Review by blacha — APPROVED** [review]

**Pull request merged**